images/kcp-binder/register.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The pipelines-service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+
+    printf "Usage: KCP_ORG="root:build-service" KCP_WORKSPACE="compute" DATA_DIR="/workspace" ./register.sh\n\n"
+
+    # Parameters
+    printf "The following parameters need to be passed to the script:\n"
+    printf "KCP_ORG: the organistation for which the workload clusters need to be registered, i.e.: root:build-service\n"
+    printf "KCP_WORKSPACE: the name of the workspace where the workload clusters get registered (created if it does not exist), i.e: compute\n"
+    printf "DATA_DIR: the location of the cluster files\n"
+    printf "          a single file with extension kubeconfig is expected in the subdirectory: gitops/credentials/kubeconfig/kcp\n"
+    printf "          kubeconfig files for compute clusters are expected in the subdirectory: gitops/credentials/kubeconfig/compute \n\n"
+}
+
+prechecks () {
+    KCP_ORG=${KCP_ORG:-}
+    if [[ -z "${KCP_ORG}" ]]; then
+        printf "KCP_ORG not set\n\n"
+        usage
+        exit 1
+    fi
+
+    KCP_ORG=${KCP_WORKSPACE:-}
+    if [[ -z "${KCP_WORKSPACE}" ]]; then
+        printf "KCP_WORKSPACE not set\n\n"
+        usage
+        exit 1
+    fi
+    
+    DATA_DIR=${DATA_DIR:-}
+    if [[ -z "${DATA_DIR}" ]]; then
+        printf "DATA_DIR not set\n\n"
+        usage
+	exit 1
+    fi
+}
+
+# populate kcp_kcfg with the location of the kubeconfig for connecting to kcp
+kcp_kubeconfig() {
+    if files=($(ls $DATA_DIR/gitops/credentials/kubeconfig/kcp/*.kubeconfig 2>/dev/null)); then
+        if [ ${#files[@]} -ne 1 ]; then
+            printf "A single kubeconfig file is expected at %s" "$DATA_DIR/gitops/credentials/kubeconfig/kcp\n"
```

Comment:
Why?
The model is that workload clusters get added to a workspace of an organization that provides compute to other workspaces through the location API.
If there are different personas managing the different organizations there should be different repos. If they are the same personas wanting to use the same repos not this script but the PipelineRun would need to be amended so that the correct org is passed by.
If we want to move away from having the ORG as a parameter but that it gets deduced from the repository structure having `kubeoncig/$tenant.$workspace.yaml`would not be sufficient. We would need to group the kcp kubeconfig/org with the compute kubeconfigs. It means that an additional level `organization` would need to be added to the hierarchy, possibly between `credentials` and `kubeconfigs`. This means that we would not allow customization of environment per organization if they are managed from the same repo, which may be acceptable.
Different environments (production, staging, etc) means different repos as discussed previously.

By any way. The structure of the repo is a different workstream. I am open to changes but I feel that these changes and associated discussions should happen outside of this PR